lib/jekyll/convertible.rb

```diff
@@ -48,8 +48,10 @@ def read_yaml(base, name, opts = {})
         end
       rescue SyntaxError => e
         Jekyll.logger.warn "YAML Exception reading #{filename}: #{e.message}"
+        raise e if self.site.config["strict_front_matter"]
       rescue => e
         Jekyll.logger.warn "Error reading file #{filename}: #{e.message}"
+        raise e if self.site.config["strict_front_matter"]
```

Comment:
Cool. Saw that, but wasn't sure where `Jekyll::Document#read` was used so I left it off. Commit to add it is up now.